src/coreclr/src/pal/src/safecrt/wcslwr_s.cpp

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/***
+*wcslwr_s.cpp - contains _wcslwr_s() routine
+*
+*
+*Purpose:
+*   _wcslwr_s converts, in place, any upper case letters in input string to
+*   lowercase.
+*
+*******************************************************************************/
+
+#include <wctype.h>
+#include <string.h>
+#include <errno.h>
+#include <limits.h>
+#include "internal_securecrt.h"
+
+#include "mbusafecrt_internal.h"
+
+DLLEXPORT errno_t __cdecl _wcslwr_s(char16_t *string, size_t sz)
+{
+    _VALIDATE_RETURN_ERRCODE(string != NULL, EINVAL);
+    if (PAL_wcsnlen(string, sz) >= sz)
+    {
+        _RETURN_DEST_NOT_NULL_TERMINATED(string, sz);
+    }
+
+    for (int i = 0; string[i] != 0; i++)
+    {
+        string[i] = towlower(string[i]);
+    }
+
+    _FILL_STRING(string, sz, sz + 1);
```

Comment:
The fill should fill all characters after the terminating zero till the end of the buffer. That's what windows implementation does. So it would require using the result from the PAL_wcsnlen + 1 as the last parameter here.

Reply:
In practice, wcsnlen also returns string upto first NULL character. e.g. this prints `7`:

```c++
   wchar_t str[] = L"The String to End All Strings!";
   str[7] = NULL;
   printf("%zd\n", wcsnlen(str, sz));
```
I think we can just remove _FILL_STRING.

Reply:
The point was to fill the rest of the buffer after the string so that after the conversion, there is a zero terminated string followed by the filling bytes.

Reply:
Ah true, __FILL_STRING eventually calls `memset`. Updated.